tests/cupyx_tests/scipy_tests/special_tests/test_logsumexp.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+import numpy
+
+import cupy
+from cupy import testing
+import cupyx.scipy.special  # NOQA
+
+
+@testing.with_requires('scipy')
+class TestLogsumexp:
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_large_inputs(self, xp, scp, dtype):
+        a = xp.arange(400)
+        return scp.special.logsumexp(a)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_more_large_inputs(self, xp, scp, dtype):
+        a = xp.arange(10000)
+        return scp.special.logsumexp(a)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_large_numbers(self, xp, scp, dtype):
+        a = xp.array([1000, 1000], dtype=dtype)
+        return scp.special.logsumexp(a)
+
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_keep_dimensions(self, xp, scp):
+        a = xp.array([[100, 1000], [1e10, 1e-10]])
+        return scp.special.logsumexp(a, axis=-1, keepdims=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
+    def test_array_inputs(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'iu':
+            pytest.skip()
+        a = testing.shaped_random((100, 1000), xp, dtype=dtype)
+        return scp.special.logsumexp(a)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_sign_argument(self, xp, scp, dtype):
+        a = xp.array([1, 1, 1], dtype=dtype)
+        b = xp.array([1, -1, -1], dtype=dtype)
+        return scp.special.logsumexp(a, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_sign_zero(self, xp, scp, dtype):
+        a = xp.array([1, 1], dtype=dtype)
+        b = xp.array([1, -1], dtype=dtype)
+        return scp.special.logsumexp(a, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
+    def test_sign_shape(self, xp, scp, dtype):
```

Comment:
I see this does not test shape. (also following tests)
```suggestion
    def test_sign_multi_dims(self, xp, scp, dtype):
```

Reply:
nice catch!